lib/routes/objects.js

```diff
@@ -406,4 +406,95 @@ module.exports =
           res.status(204);
         }
     });
+
+    /* eslint-disable max-statements */
+    fastify.route({
+      method: 'PUT',
+      url: '/objects/:name/:env/rollback',
+      preHandler: fastify.auth([fastify.verifyAuthentication]),
+      schema: {
+        params: {
+          type: 'object',
+          properties: {
+            name: { type: 'string' },
+            env: { type: 'string' }
+          }
+        },
+        body: {
+          type: 'object',
+          properties: {
+            hops: { type: 'number', minimum: 1, maximum: 20 }
+          }
+        },
+        response: {
+          204: {}
+        }
+      },
+      handler:
+        /**
+         * @type {FastifyHandler}
+         */
+        async (req, res) => {
+          const {
+            params: { name, env },
+            body: { hops = 1 }
+          } = req;
+
+          const [obj, objVersions] = await Promise.all([
+            fastify.getObject({ name, env }),
+            fastify.getAllObjectVersions({ name, env })
+          ]);
+
+          if (!obj) {
+            throw fastify.httpErrors.notFound(
+              `Object '${name}' not found in '${env}'`
+            );
+          }
+
+          // Recursivly find version back to N hops
+          let timestamp = obj.headTimestamp;
+          let hop = 0;
+          let record;
+          while (timestamp && hop < hops + 1) {
+            record = await fastify.getHistoryRecord({ name, env, timestamp });
+            timestamp = record.prevTimestamp;
+            hop++;
+          }
+
+          if (hop < 2 || !record) {
```

Comment:
Why 2?

Reply:
Because if only 1 "hop", means we're looking at the history record associated with the current head: therefore no rollback can be done since there is no previous version.